### PR TITLE
Fixes #9923 - added recursion for nested objects

### LINKED ISSUE #9923: HTML2 generator outputs "undefined" DTOs
Hi, I am using Open API generator plugin 5.1.1.

I instrumented my projects to generate HTML2 documentation. 

It works fine for all of them but one which, for the response DTOs, prints an **undefined** statement as for the attached snapshot:

image

I opened the browser debug console and I found the following error:

```
SyntaxError: Error resolving $ref pointer "http://localhost:7001/my-api/help#/components/schemas/NickNameTypeDTO". 
Token "components" does not exist.
    at Function.syntax (help:8110)
    at Pointer.resolve (help:7898)
    at $Ref.resolve (help:7901)
    at $Refs._resolve (help:7904)
    at dereference$Ref (help:7856)
    at help:7856
    at Array.forEach (<anonymous>)
    at crawl (help:7856)
    at help:7856
    at Array.forEach (<anonymous>)
```

How can I further investigate on the causes of that issue?

Thank you so much!

### PR DESCRIPTION
**Pull request by 0x7162**
Added recursive check for the references ($ref)
Now we should support nested objects.

[x ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
[ x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
[ x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
./mvnw clean package 
./bin/generate-samples.sh
./bin/utils/export_docs_generators.sh
Commit all changed files.
This is important, as CI jobs will verify all generator outputs of your HEAD commit as it would merge with master.
These must match the expectations made by your contribution.
You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example ./bin/generate-samples.sh bin/configs/java*.
For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
[x ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

To fix #9923

### CODE REVIEW

**Comment by 0x7162** [comment]
as the code is in JS (though for documentation generation)(html2)

@CodeNinjai @frol @cliffano

**Comment by wing328** [comment]
@0x7162 thanks for the PR. Let's give it a try.

**Comment by wing328** [comment]
FYI. samples updated via 0296380f006

File: modules/openapi-generator/src/main/resources/htmlDocs2/index.mustache [hunk 1]
```diff
@@ -452,6 +452,10 @@
                                         }
                                         if (schema.$ref != null) {
                                           schema = defsParser.$refs.get(schema.$ref);
+                                          Object.keys(schema.properties).forEach( (item) => {
+                                            if (schema.properties[item].$ref != null) {
+                                              schema.properties[item] = defsParser.$refs.get(schema.properties[item].$ref);
+                                            }
```
Comment (kas1337) [inline modules/openapi-generator/src/main/resources/htmlDocs2/index.mustache:458]: @wing328 The forEach-Loop is never closed (missing `});`). I opened this issue https://github.com/OpenAPITools/openapi-generator/issues/12842

Comment (wing328) [reply modules/openapi-generator/src/main/resources/htmlDocs2/index.mustache:458]: @kas1337 thanks for catching the issue. Can you please file a PR with the suggested fix?

**Pull request merged**